1/main.go

```
@@ -0,0 +1,184 @@
+	if r.Method != http.MethodGet {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		fmt.Fprintln(w, "Invalid method, use GET")
+		return
+	}
+	token, err := tokenFromRequest(r)
+	if err != nil {
+		w.WriteHeader(http.StatusUnauthorized)
+		fmt.Fprintln(w, err.Error())
+		return
+	}
+	_, err = db.Exec(`delete from sessions where token = ?1;`, token)
+	// at this point, user has no idea if his token was valid before his
+	// request to deauthorization, but at least he can be sure that supplied
+	// token is no longer valid after receiving HTTP 200 query reply
+	if err != nil {
+		log.Printf("Failed to delete access token '%s' from database: %s\n", token, err.Error())
+		w.WriteHeader(http.StatusInternalServerError)
+	}
+}
+
+func profileHandler(db *sql.DB, w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		fmt.Fprintln(w, "Invalid method, use GET")
+		return
+	}
+	token, err := tokenFromRequest(r)
+	if err != nil {
+		w.WriteHeader(http.StatusUnauthorized)
+		fmt.Fprintln(w, err.Error())
+		return
+	}
+	var name string
+	var password string
+	row := db.QueryRow(`select name, password from (select * from users join sessions) where token = ?;`, token)
+	if err := row.Scan(&name, &password); err != nil {
+		log.Println("Failed to query user data from database", err.Error())
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	fmt.Fprintf(w, "Your name is %s and your password is %s\n", name, password)
+}
+
+func tokenFromRequest(r *http.Request) (string, error) {
+	// here I assume that r.Header values (arrays of strings) always have at least one element,
+	// documentation does not tell anything about it: https://golang.org/pkg/net/http/#Header
+	cookie, exists := r.Header["Cookie"]
+	if !exists {
+		return "", errors.New("Cookie with access token must be supplied")
+	}
+	var token string
+	fmt.Sscanf(cookie[0], "sessionToken=%s", &token)
```

Ignoring the error here led you to ignore the non-working code.

Error from `Sscanf()` -> `input does not match format`

---

Fixed.